[stable/falco] Add GCloud PubSub integration

tembleking opened the pull request:
* Add GCloud PubSub integration

This allows Falco to publish alerts to a PubSub topic

Signed-off-by: Federico Barcelona <fede.barcelona@sysdig.com>

tembleking committed:
[stable/falco] Add GCloud PubSub integration

* Add GCloud PubSub integration

This allows Falco to publish alerts to a PubSub topic

Signed-off-by: Federico Barcelona <fede_rico_94@hotmail.com>

nestorsalceda commented on stable/falco/values.yaml:
```diff
@@ -199,6 +199,21 @@ integrations:
+  #     filename: /var/run/falco/nats
+  pubsubOutput:
+    enabled: false
+    credentials_data: ""
```
Follow naming conventions: https://helm.sh/docs/chart_best_practices/#naming-conventions

The AWS ones should be also camelCase, that was a violation that will be fixed soon.

nestorsalceda commented on stable/falco/templates/secret.yaml:
```diff
@@ -1,8 +1,9 @@
 kind: Secret
 metadata:
-  name: {{ template "falco.fullname" . }}
+  name: {{ template "falco.fullname" . }}-sns
```
Use only one secret instead of creating two different objects.

nestorsalceda commented on stable/falco/templates/secret.yaml:
```diff
@@ -13,3 +14,20 @@ data:
+    heritage: "{{ .Release.Service }}"
+type: Opaque
+data:
+  credentials-data: {{ .Values.integrations.pubsubOutput.credentials_data | toJson | b64enc | quote }}
```
Put these entries under a `gcp` entry and the AWS ones under `aws`

nestorsalceda commented on stable/falco/templates/daemonset.yaml:
```diff
@@ -113,20 +113,40 @@ spec:
+            - name: GOOGLE_PROJECT_ID
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "falco.fullname" . }}-pubsub
```
There's no need of `-pubsub` suffix

nestorsalceda commented on stable/falco/templates/daemonset.yaml:
```diff
@@ -113,20 +113,40 @@ spec:
+            - name: GOOGLE_CREDENTIALS_DATA
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "falco.fullname" . }}-pubsub
```
There's no need of `-pubsub` suffix, we will use the key entry.

nestorsalceda commented on stable/falco/templates/daemonset.yaml:
```diff
@@ -113,20 +113,40 @@ spec:
               valueFrom:
                 secretKeyRef:
-                  name: {{ template "falco.fullname" . }}
+                  name: {{ template "falco.fullname" . }}-sns
```
There's no need of `-sns` suffix

nestorsalceda commented on stable/falco/templates/daemonset.yaml:
```diff
@@ -113,20 +113,40 @@ spec:
               valueFrom:
                 secretKeyRef:
-                  name: {{ template "falco.fullname" . }}
+                  name: {{ template "falco.fullname" . }}-sns
```
There's no need of `-sns` suffix

nestorsalceda reviewed (changes requested):
That's nice work! 

We just need to polish some details and remember to add the new parameters in the README.md for documentation purposes.

Thank  you!

tembleking committed:
[stable/falco] Fix values to follow naming conventions

Signed-off-by: Federico Barcelona <fede_rico_94@hotmail.com>

tembleking committed:
[stable/falco] Changes requested in the PR

- Follow naming conventions
- Use only one secret instead of two different ones

Signed-off-by: Federico Barcelona <fede_rico_94@hotmail.com>
